float32 depth
float32 velocity
float32 zero_depth_pressure
